cpp/src/arrow/util/align_util_test.cc

```diff
@@ -145,6 +150,84 @@ TEST(BitmapWordAlign, UnalignedDataStart) {
   CheckBitmapWordAlign<8>(P, 1017, 64, {63, 1, 1080, A, 0, 0});
   CheckBitmapWordAlign<8>(P, 1017, 128, {63, 1, 1144, A + 128, 64, 1});
 }
-
 }  // namespace internal
+
+TEST(EnsureAlignment, Array) {
+  MemoryPool* pool = default_memory_pool();
+  auto rand = ::arrow::random::RandomArrayGenerator(1923);
+  auto random_array = rand.UInt8(/*size*/ 50, /*min*/ 0, /*max*/ 100,
+                                 /*null_probability*/ 0, /*alignment*/ 512, pool);
+  ASSERT_OK_AND_ASSIGN(auto aligned_array,
```

Comment:
Maybe assert that `CheckAlignment(..., 1024)` returns `false` before you call `EnsureAlignment`?

Reply:
Made the change, thanks.